helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java

```diff
@@ -104,11 +107,18 @@ public AssignmentResult() {
     Map<String, String> swapInstances;
   }
 
+  private enum AssignmentFormat{
+    IdealStateFormat,
+    CurrentStateFormat
```

Comment:
TFTR. Normally it is advised to have enum with this convention. However, when we want to access enum name, isn't this format better than having enum with value? (Or at lease keep all enums in REST in the same format? Like how ClusterAccessor.java and other REST resources do it.)

Reply:
I vote to ensure all enums in the project keep the same style. I know rest has some legacy namings that not necessarily follow the rule. But we should not add more.

Regarding which name is better, you can add a string field for the Enum object if really needed. But they make not much difference from my perspective. It is up to you.

Reply:
We had an offline discussion and decide to keep this way for the following reason:
1. Change enum name to uppercase like `IDEAL_STATE_FORMAT` and keep the same format in input JSON. This is not optimal because it will impact user readability in input.
2. Because we are doing auto converting from JSON input, adding a string field for the Enum will require more code change like comparing the string many times and have dedicated sanity check code.